cfme/physical/physical_rack.py

```diff
@@ -0,0 +1,336 @@
+        """ Select all physical rack objects """
+        physical_racks = list(physical_racks)
+        checked_physical_racks = list()
+        view = navigate_to(self, 'All')
+
+        for physical_rack in physical_racks:
+            view.entities.get_entity(name=physical_rack.name, surf_pages=True).check()
+            checked_physical_racks.append(physical_rack)
+        return view
+
+    def all(self):
+        """returning all physical_racks objects"""
+        physical_rack_table = self.appliance.db.client['physical_racks']
+        ems_table = self.appliance.db.client['ext_management_systems']
+        physical_rack_query = (
+            self.appliance.db.client.session
+                .query(physical_rack_table.name, ems_table.name)
+                .join(ems_table, physical_rack_table.ems_id == ems_table.id))
+        provider = None
+
+        if self.filters.get('provider'):
+            provider = self.filters.get('provider')
+            physical_rack_query = physical_rack_query.filter(ems_table.name == provider.name)
+        physical_racks = []
+        for name, ems_name in physical_rack_query.all():
+            physical_racks.append(self.instantiate(name=name,
+                                                   provider=provider or get_crud_by_name(ems_name)))
+        return physical_racks
+
+
+class ComputePhysicalInfrastructureRacksView(BaseLoggedInPage):
+    """Common parts for rack views."""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
+
+    @property
+    def in_compute_physical_infrastructure_racks(self):
+        return (self.logged_in_as_current_user and
+                self.navigation.currently_selected == ["Compute", "Physical Infrastructure",
+                                                       "Racks"])
+
+
+class PhysicalRackQuadIconEntity(BaseQuadIconEntity):
+    @property
+    def data(self):
+        return {
+            'no_physical_servers': int(self.browser.text(self.QUADRANT.format(pos="a"))),
+        }
+
+
+class PhysicalRackTileIconEntity(BaseTileIconEntity):
+    quad_icon = ParametrizedView.nested(PhysicalRackQuadIconEntity)
+
+
+class PhysicalRackListEntity(BaseListEntity):
+    pass
+
+
+class NonJSPhysicalRackEntity(NonJSBaseEntity):
+    quad_entity = PhysicalRackQuadIconEntity
+    list_entity = PhysicalRackListEntity
+    tile_entity = PhysicalRackTileIconEntity
+
+
+class PhysicalRackEntity(JSBaseEntity):
+    @property
+    def data(self):
+        data_dict = super(PhysicalRackEntity, self).data
+        if 'quadicon' in data_dict and data_dict['quadicon']:
+            quad_data = document_fromstring(data_dict['quadicon'])
+            data_dict['no_host'] = int(quad_data.xpath(self.QUADRANT.format(pos="a"))[0].text)
+        return data_dict
+
+
+class PhysicalRackDetailsToolbar(View):
+    """Represents physical rack toolbar and its controls."""
+    configuration_button = Dropdown(text="Configuration")
```

Comment:
just `configuration` here, as _button is misleading, and in general we avoid these widget suffixes.